src/thunderbots/software/ai/primitive/movespin_primitive.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "ai/primitive/primitive.h"
+#include "geom/angle.h"
+#include "geom/point.h"
+
+class MoveSpinPrimitive : public Primitive
+{
+   public:
+    static const std::string PRIMITIVE_NAME;
+    /**
+     * Creates a new MoveSpin Primitive
+     *
+     * @param robot_id The id of the Robot to run this Primitive
+     * @param dest The final destination of the movement
+     * @param angular_vel The angular velocity of the robot
+     * of the movement
+     */
+    explicit MoveSpinPrimitive(unsigned int robot_id, const Point &dest,
+                               const AngularVelocity &angular_vel);
+
+    /**
+     * Creates a new MoveSpin Primitive from a Primitive message
+     *
+     * @param primitive_msg The message from which to create the Move Primitive
+     */
+    explicit MoveSpinPrimitive(const thunderbots_msgs::Primitive &primitive_msg);
+
+    std::string getPrimitiveName() const override;
+
+    unsigned int getRobotId() const override;
+
+    /**
+     * Gets the robot's destination
+     *
+     * @return The robot's destination as a Point(X,Y)
+     */
+    Point getDestination() const;
+
+    /**
+     * Gets the robot's angular velocity in rad/s
+     *
+     * @return The robot's angular velocity in rad/s
+     */
+    AngularVelocity getAngularVelocity() const;
+
+    std::vector<double> getParameters() const override;
+
+    std::vector<bool> getExtraBits() const override;
```

Comment:
`getParameters` and `getExtraBits` should have descriptive javadoc comments, see other primitives for examples.

Reply:
I don't think it needs one. This is overriding a function, and the commend documentation is there. We typically haven't been re-commenting overridden functions.

Reply:
Generally I agree, but I think it's important that these functions have explicit documentation for the form of the vector that they're returning (or if those vectors are empty) because it's otherwise not indicated what the semantic meaning of each element in the vector means. See other primitives for examples.

Reply:
Ok fair point. I agree